### PR TITLE
Update samples to support latest Google Cloud Python

### PR DESCRIPTION
**Pull request by jonparrott**
(I recommend reviewing commit by commit)

### CODE REVIEW

File: datastore/cloud-client/snippets.py [hunk 1]
```diff
@@ -362,9 +362,12 @@ def cursor_paging(client):
     def get_one_page_of_tasks(cursor=None):
         query = client.query(kind='Task')
         query_iter = query.fetch(start_cursor=cursor, limit=5)
-        tasks, _, cursor = query_iter.next_page()
+        page = next(query_iter.pages)
 
-        return tasks, cursor
+        tasks = list(page)
+        next_cusor = query_iter.next_page_token
```
Comment (jerjou) [inline datastore/cloud-client/snippets.py]: `next_cusor`? :)

File: nox.py [hunk 2]
```diff
@@ -317,9 +319,19 @@ def mark_if_necessary(requirement):
...
+            if requirement.startswith(library):
+                return True
+
     requirements = [
         mark_if_necessary(requirement) for requirement in requirements]
 
+    requirements = [
+        requirement for requirement in requirements if not
+        is_ignored(requirement)]
+
```
Comment (jerjou) [inline nox.py]: Could put this conditional in the previous comprehension, no?

File: nox.py [hunk 3]
```diff
@@ -317,9 +319,19 @@ def mark_if_necessary(requirement):
                     requirement.strip())
         return requirement
 
+    def is_ignored(requirement):
+        """Ignores certain libraries."""
+        for library in IGNORED_LIBRARIES:
+            if requirement.startswith(library):
+                return True
```
Comment (jerjou) [inline nox.py]: Could do this more succinctly:

    return any(requirement.startswith(l) for l in IGNORED_LIBRARIES)

File: datastore/cloud-client/snippets.py (hunk 1 above)
Comment (jonparrott) [inline datastore/cloud-client/snippets.py]: Fixed.

File: nox.py (hunk 2 above)
Comment (jonparrott) [inline nox.py]: I'd rather not.

File: nox.py (hunk 3 above)
Comment (jonparrott) [inline nox.py]: I like it, but it's almost too clever. I'm going to leave as-is for now.

Comment (waprin) [inline nox.py]: Don't care either way but don't think it's  too clever, a bit more English-esque if anything.

**Comment by waprin** [comment]
LGTM, shoutout to @dhermes for a lot of nice refactors that make these samples look a lot nicer.

**Comment by jonparrott** [comment]
@dpebot merge when travis passes, please.

File: bigquery/cloud-client/snippets.py [hunk 4]
```diff
@@ -157,19 +125,10 @@ def list_rows(dataset_name, table_name, project=None):
...
-        results, total_rows, page_token = table.fetch_data(
-            max_results=25, page_token=page_token)
-        rows.extend(results)
-
-        if not page_token:
-            break
+    # Load at most 25 results. You can change the max_results argument to load
+    # more rows from BigQuery, but note that this can take some time. It's
+    # preferred to use a query.
+    rows = table.fetch_data(max_results=25)
```
Comment (dhermes) [inline bigquery/cloud-client/snippets.py]: You may prefer to just consume the iterator right here:

```python
rows = list(table.fetch_data(max_results=25))
```

There is too much suspense between this and the `for row in rows` (when the request(s) actually get made)

Comment (jonparrott) [inline bigquery/cloud-client/snippets.py]: Fair enough, done.

**Comment by Zayanusman221** [comment]
cf1b498536c3f28cbd6c7d58d665655cdfaf571d

**Pull request merged**